lib/sinon/sandbox.js

```diff
@@ -95,6 +95,10 @@ extend(sinonSandbox, {
     },
 
     restore: function () {
+        if (arguments.length) {
+            throw new Error("sandbox.restore() does not take any parameters. Perhaps you meant stub.restore()");
+        }
+
         sinonCollection.restore.apply(this, arguments);
```

Comment:
Good point, this does seem a bit weird. That line has been there since restore was first introduced for the sandbox in e1bcec820, so maybe @marcushg remembers if this was intentional or just done by convention.